tracer/src/Datadog.Trace/ISpanContextExtractor.cs

```diff
@@ -14,17 +14,31 @@ namespace Datadog.Trace
     /// The ISpanContextExtractor is responsible for extracting SpanContext in the rare cases where the Tracer couldn't propagate it itself.
     /// This can happen for instance when libraries add an extra layer above the instrumented ones
     /// (eg consuming Kafka messages and enqueuing them prior to generate a span).
-    /// When enabled (and present in the headers) also used to set data streams monitoring checkpoints.
+    /// When messageType and target are specified, also used to set data streams monitoring checkpoints (if enabled).
     /// </summary>
     public interface ISpanContextExtractor
     {
         /// <summary>
-        /// Given a SpanContext carrier and a function to access the values, this method will extract the SpanContext if any.
+        /// Given a SpanContext carrier and a function to access the values, this method will extract SpanContext if any
         /// </summary>
         /// <param name="carrier">The carrier of the SpanContext. Often a header (http, kafka message header...)</param>
         /// <param name="getter">Given a key name, returns values from the carrier</param>
         /// <typeparam name="TCarrier">Type of the carrier</typeparam>
         /// <returns>A potentially null Datadog SpanContext</returns>
         public ISpanContext? Extract<TCarrier>(TCarrier carrier, Func<TCarrier, string, IEnumerable<string?>> getter);
+
+        /// <summary>
+        /// Given a SpanContext carrier and a function to access the values, this method will extract the SpanContext
+        /// and the PathwayContext, and will set a DataStreams Monitoring checkpoint if enabled.
+        /// You should only call <see cref="ExtractIncludingDsm{TCarrier}"/> once on the message <paramref name="carrier"/>. Calling
+        /// multiple times may lead to incorrect stats when using Data Streams Monitoring.
+        /// </summary>
+        /// <param name="carrier">The carrier of the SpanContext. Often a header (http, kafka message header...)</param>
+        /// <param name="getter">Given a key name, returns values from the carrier</param>
+        /// <param name="messageType">For Data Streams Monitoring: The type of messaging system where the message is coming from.</param>
+        /// <param name="source">For Data Streams Monitoring: The queue or topic where the message is coming from.</param>
+        /// <typeparam name="TCarrier">Type of the carrier</typeparam>
+        /// <returns>A potentially null Datadog SpanContext</returns>
+        public ISpanContext? ExtractIncludingDsm<TCarrier>(TCarrier carrier, Func<TCarrier, string, IEnumerable<string?>> getter, string messageType, string source);
```

Comment:
Just pointing out this is a breaking change, which is obviously fine to do in v3. I'm pointing it out, because this is your _one_ chance to add to this interface this year, so make sure it's definitely what you want - once you have it, you're stuck with it 😅 

Which _actually_ makes me thing we shouldn't even have this interface if _we_ don't need it 🤔 Customers can easily create a wrapper around the extractors, so it doesn't add much at all, just means we _can't_ evolve the `SpanContextExtractor` code 🤔

Reply:
_we_ don't need it indeed, we offer it for clients who'd want to inject span context & get DSM data from non-instrumented systems.
And there is a bunch of methods in there that are internal, so this is not a handy helper, it's doing stuff that the clients can't do themselves 🤔

Reply:
Just discussed with the team and yeah, tl;dr; we should just delete this interface 😄 I'll do that in a separate PR, you can just revert the `ISpanContextExtractor` and `ISpanContextInjector` changes 🙂

Reply:
Injector has no interface, what you mean is that the Extractor shouldn't have an interface either, we should just offer the class itself directly ?

Reply:
> Injector has no interface, what you mean is that the Extractor shouldn't have an interface either, we should just offer the class itself directly ?

Oh, yes, exactly. We'll delete `ISpanContextExtractor`, and the customer will just use `SpanContextExtractor` directly (which they have to do anyway, there's basically little to no reason for them to use `ISpanContextExtractor` currently)